src/Credentials/Utils/Validator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Aws\Credentials\Utils;
+
+final class Validator
+{
+    private function __construct() {}
+
+    public static function validateURI(
```

Comment:
I think this could be moved to the InstanceProfileProvider class.  validation will vary depending on the provider (see the EcsCredentialProvider)

Reply:
We won't need it to accept arguments aside from the Uri itself.